client/jetpack-cloud/sections/partner-portal/company-details-form/index.tsx

```diff
@@ -0,0 +1,193 @@
+	const submitNotificationId = 'company-details-form';
+	const showCountryFields = countryOptions.length > 0;
+
+	const [ isSubmitting, setIsSubmitting ] = useState( false );
+	const [ name, setName ] = useState( partner?.name ?? '' );
+	const [ country, setCountry ] = useState( partner?.address.country ?? '' );
+	const [ city, setCity ] = useState( partner?.address.city ?? '' );
+	const [ line1, setLine1 ] = useState( partner?.address.line1 ?? '' );
+	const [ line2, setLine2 ] = useState( partner?.address.line2 ?? '' );
+	const [ postalCode, setPostalCode ] = useState( partner?.address.postal_code ?? '' );
+	const [ addressState, setAddressState ] = useState( partner?.address.state ?? '' );
+
+	const stateOptions = stateOptionsMap.hasOwnProperty( country )
+		? stateOptionsMap[ country ]
+		: false;
+
+	const updateCompanyDetails = useUpdateCompanyDetailsMutation( {
+		onSuccess: () => {
+			dispatch(
+				successNotice( translate( 'Company details has been updated' ), {
+					id: submitNotificationId,
+				} )
+			);
+			setIsSubmitting( false );
+		},
+		onError: ( error: APIError ) => {
+			dispatch(
+				errorNotice( error.message, {
+					id: submitNotificationId,
+				} )
+			);
+			setIsSubmitting( false );
+		},
+	} );
+
+	const handleSubmit = useCallback(
+		( e ) => {
+			e.preventDefault();
+			setIsSubmitting( true );
+			dispatch( removeNotice( submitNotificationId ) );
+
+			const payload = {
+				name: name,
+				city: city,
+				line1: line1,
+				line2: line2,
+				country: country,
+				postal_code: postalCode,
+				state: addressState,
+			};
+
+			updateCompanyDetails.mutate( payload );
+
+			dispatch(
+				recordTracksEvent( 'calypso_partner_portal_update_company_details_submit', {
+					partner_id: partner?.id,
+					...payload,
+				} )
+			);
+		},
+		[ dispatch, updateCompanyDetails ]
+	);
+
+	return (
+		<Card className="company-details-form">
+			<form onSubmit={ handleSubmit }>
+				<FormFieldset>
+					<FormLabel htmlFor="name">{ translate( 'Company name' ) }</FormLabel>
+					<FormTextInput
+						id="name"
+						name="name"
+						value={ name }
+						onChange={ ( event: any ) => setName( event.target.value ) }
+						disabled={ isSubmitting }
+					/>
+				</FormFieldset>
+				<FormFieldset>
+					<FormLabel>{ translate( 'Country' ) }</FormLabel>
+					{ showCountryFields && (
+						<SelectDropdown
+							className="company-details-form__dropdown"
+							initialSelected={ country }
+							options={ countryOptions }
+							onSelect={ ( option: any ) => {
+								setCountry( option.value );
+								// Reset the value of state since it no longer matches with the selected country.
+								setAddressState( '' );
+							} }
+							disabled={ isSubmitting }
+							isLoading={ countryOptions.length === 0 }
+						/>
+					) }
+
+					{ ! showCountryFields && <TextPlaceholder /> }
+				</FormFieldset>
+				<FormFieldset>
+					{ showCountryFields && stateOptions && (
```

Comment:
There's some extra spacing when the field is not visible because of the fieldset:
![Screen Shot 2022-05-19 at 21 03 45](https://user-images.githubusercontent.com/22746396/169368706-1b54b923-3f88-45f7-8e55-94a655be421f.png)